### DISCUSSION TITLE
Coffee-editor modifying the .coffee model file and clicking save has no effect. The refresh is still the previous value.

### DISCUSSION THREAD

**Discussion by 136861535** [eclipse-emfcloud/emfcloud#143] [Coffee Editor]
Coffee-editor modifying the .coffee model file and clicking save has no effect. The refresh is still the previous value.why?

**Comment by eneufeld** [comment]
That sounds like something is not working as expected. Can you check your browser console log and your backend console and paste the errors logged there?

**Reply by 136861535** [reply]
![图片](https://user-images.githubusercontent.com/101307516/158506948-247ff0e0-cdf1-4b82-a604-e38584843a1c.png)
I fill in an "a" here, and then Ctrl + s to save it。The following is the log information：
![图片](https://user-images.githubusercontent.com/101307516/158507275-e2120b07-855b-4125-903d-fbcce1d42a4a.png)
After refreshing the interface, this value is still empty。
![图片](https://user-images.githubusercontent.com/101307516/158507391-1a73f00c-7aae-415a-a14a-17f67bf3f57a.png)

This method is executed when saving：
![图片](https://user-images.githubusercontent.com/101307516/158508184-384dc292-0866-4ffa-a46e-3049706cee33.png)

**Reply by 136861535** [reply]
Please answer it, thank you!

**Reply by eneufeld** [reply]
I need to debug this on my side and I will come back to you later.

**Reply by 136861535** [reply]
![图片](https://user-images.githubusercontent.com/101307516/158562450-8d57fc54-934d-4a3e-8fd0-88a9efcdac1c.png)
[](https://fanyi.baidu.com/translate?aldtype=16047&query=docker%E8%BF%90%E8%A1%8Ccoffee-editor%E6%97%B6%E6%8A%A5%E9%94%99&keyfrom=baidu&smartresult=dict&lang=auto2zh###)[](javascript:void(0);)Because we reported an error when running on windows, we modified the code in the above file, as shown in the figure.

We also replaced the following file (coffeemodelservermodule. Java):
```
public class CoffeeModelServerModule extends EMSNotationModelServerModule {
	
	protected void configure() {
	      super.configure();
	      bind(ServerController.class).to(bindServerController()).in(Singleton.class);
	   }

	@Override
	protected Class<? extends ModelResourceManager> bindModelResourceManager() {
		return CoffeeModelResourceManager.class;
	}

	 protected Class<? extends ServerController> bindServerController() {
	      return WindowDefaultServerController.class;
	   }
	 
	@Override
	protected void configureEPackages(final MultiBinding<EPackageConfiguration> binding) {
		super.configureEPackages(binding);
		binding.add(CoffeePackageConfiguration.class);
	}

	@Override
	protected void configureCodecs(final MapBinding<String, Codec> binding) {
		super.configureCodecs(binding);
		binding.put(CoffeeResource.FILE_EXTENSION, CoffeeCodec.class);
		binding.put(ModelServerPathParameters.FORMAT_JSON, CoffeeTreeJsonCodec.class);
	}

	@Override
	protected void configureCommandCodecs(MapBinding<String, CommandContribution> binding) {
		super.configureCommandCodecs(binding);

		// Nodes
		binding.put(AddManualTaskCommandContribution.TYPE, AddManualTaskCommandContribution.class);
		binding.put(AddAutomatedTaskCommandContribution.TYPE, AddAutomatedTaskCommandContribution.class);
		binding.put(AddDecisionNodeCommandContribution.TYPE, AddDecisionNodeCommandContribution.class);
		binding.put(AddMergeNodeCommandContribution.TYPE, AddMergeNodeCommandContribution.class);
		binding.put(RemoveNodeCommandContribution.TYPE, RemoveNodeCommandContribution.class);

		// Flows (Edges)
		binding.put(AddFlowCommandContribution.TYPE, AddFlowCommandContribution.class);
		binding.put(AddWeightedFlowCommandContribution.TYPE, AddWeightedFlowCommandContribution.class);
		binding.put(RemoveFlowCommandContribution.TYPE, RemoveFlowCommandContribution.class);
		binding.put(SetFlowSourceCommandContribution.TYPE, SetFlowSourceCommandContribution.class);
		binding.put(SetFlowTargetCommandContribution.TYPE, SetFlowTargetCommandContribution.class);
	}

}
```

Do you think this is the reason?

**Reply by eneufeld** [reply]
based on your screenshots I would claim you are not using the current master. Could you try using it?
Whether your changes broke something is hard to say. I would need to duplicate those.
I tested the current master on linux and I could modify the data and save it.
There might be issues on windows with path. Most of these issues should be fixed in the latest versions of the modelserver which we are currently updating to.

**Reply by 136861535** [reply]
Based on your screenshots I would claim you are not using the current master, What does this sentence mean? How should I try? Should I restore the code?

**Reply by eneufeld** [reply]
Yes, could you try reset your code and pull the current master branch from https://github.com/eclipse-emfcloud/coffee-editor/

**Reply by 136861535** [reply]
OK, thank you！
